src/support/World.coffee

```diff
@@ -0,0 +1,20 @@
+rek = require 'rekuire'
+chai = require 'chai'
+chaiAsPromised = require 'chai-as-promised'
+ElementHelper = rek 'ElementHelper'
+Transform = rek 'Transform'
+
+class World
+  Q: require 'q'
+  elementHelper: new ElementHelper()
+  transform: new Transform()
+  currentPage: null
+  pageObjectMap: null
+
+  constructor: (callback) ->
+    chai.use chaiAsPromised
+    @expect = chai.expect
+
+    callback()
```

Comment:
Great question @watsoncj . I think the key thing to note is that Cucumber.js tries to instantiate the World for you by calling the constructor each time a new scenario begins. It passes a callback to the constructor and expects you to call the callback to indicate that you're ready to move on. I had originally thought this was just provided on the off-chance that you needed to support asynchronous function calls in your constructor. However, I reread the [World section](https://github.com/cucumber/cucumber-js#world) of the Cucumber.js README and realized that the callback function can provide benefits even if you're calling it synchronously. The second example in that section shows that you can pass an object to the callback function, which tells Cucumber.js to use that object as the World instead of whatever may have been instantiated by the constructor function. This functionality in Cucumber.js opens up some possibilities that I didn't realize existed and subsequently aren't supported by CukeFarm. I'll open an issue to investigate whether this should be supported by CukeFarm.